types.go

```diff
@@ -11,11 +11,30 @@ import (
 	"github.com/filecoin-project/specs-storage/storage"
 )
 
-type Piece struct {
-	DealID *abi.DealID
+// PieceWithOptionalDealInfo is a tuple of piece and deal info
+type PieceWithDealInfo struct {
```

Comment:
I'd remove this type, and only keep PieceWithOptionalDealInfo.

The only place we use this in is SealPiece, where we can either just use PieceWithOptionalDealInfo, and ensure that DealInfo is set, or just pass 2 separate params.